crate/server/src/start_kms_server.rs

```diff
@@ -36,8 +48,168 @@ use crate::{
+/// 4. If no key exists and no migration key is available, a new RSA keypair is created and stored.
+///
+/// This ensures that the KMS either reuses an existing RSA keypair, imports one for migration purposes,
+/// or generates a new one, depending on availability and configuration.
+///
+/// # Arguments
+///
+/// * `kms_server` - A reference-counted pointer to the KMS instance, used to interact with the key management backend.
+/// * `server_params` - A reference-counted pointer to the server configuration, which may contain a migration key.
+///
+/// # Errors
+///
+/// Returns a `KmsError` if:
+/// * The RSA keypair fetch, import, or creation fails.
+/// * The migration PEM key is malformed or cannot be parsed.
+/// * Conversion between KMIP and OpenSSL formats fails.
+pub async fn handle_google_cse_rsa_keypair(
+    kms_server: &Arc<KMS>,
+    server_params: &Arc<ServerParams>,
+) -> KResult<()> {
+    let uid_sk = format!("{GOOGLE_CSE_ID}_rsa");
+    let uid_pk = format!("{GOOGLE_CSE_ID}_rsa_pk");
+
+    let get_request = Get {
+        unique_identifier: Some(UniqueIdentifier::TextString(uid_sk.clone())),
+        key_format_type: Some(KeyFormatType::PKCS1),
+        key_wrap_type: Some(KeyWrapType::NotWrapped),
+        key_compression_type: None,
+        key_wrapping_specification: None,
+    };
+
+    match kms_server.get(get_request, "admin", None).await {
+        Ok(resp) if matches!(resp.object_type, ObjectType::PrivateKey) => {
+            info!("RSA Keypair for Google CSE already exists.");
+            return Ok(());
+        }
+        Ok(resp) => {
+            return Err(KmsError::CryptographicError(format!(
+                "Unexpected object type for Google CSE RSA keypair: {:?}",
+                resp.object_type
+            )));
+        }
+        Err(_) => {
+            info!("RSA Keypair for Google CSE not found from existing DB.");
+        }
+    }
+
+    if let Some(migration_key_pem) = &server_params.google_cse.google_cse_migration_key {
+        info!("Found Google CSE migration key, importing it.");
+
+        let key_bytes = pem::parse(migration_key_pem)
+            .map_err(|e| {
+                KmsError::CryptographicError(format!(
+                    "Error parsing google_cse_migration PEM key: {e}"
+                ))
+            })?
+            .contents()
+            .to_vec();
+
+        // Build PrivateKey object
+        let object_sk = Object::PrivateKey(PrivateKey {
+            key_block: KeyBlock {
+                key_format_type: KeyFormatType::PKCS8,
+                key_compression_type: None,
+                key_value: Some(KeyValue::Structure {
+                    key_material: KeyMaterial::ByteString(key_bytes.into()),
+                    attributes: Some(Attributes::default()),
+                }),
+                cryptographic_algorithm: None,
+                cryptographic_length: None,
+                key_wrapping_data: None,
```

Comment:
probably regroup None params:

```suggestion
                key_value: Some(KeyValue::Structure {
                    key_material: KeyMaterial::ByteString(key_bytes.into()),
                    attributes: Some(Attributes::default()),
                }),
                ..Default::default()
```

Reply:
the trait bound `cosmian_kmip::kmip_2_1::kmip_data_structures::KeyBlock: std::default::Default` is not satisfied
the trait `std::default::Default` is not implemented for `cosmian_kmip::kmip_2_1::kmip_data_structures::KeyBlock`

Reply:
There is no default on KeyBlock

Reply:
Ok right. Implementing Default on `KeyBlock` needs to set #[default] on KMIP enum. Could be done (but with a price).